platform/mbed_retarget.h

```diff
@@ -21,15 +21,18 @@
 
 #if __cplusplus
 #include <cstdio>
+#else
+#include <stdio.h>
```

Comment:
huh, out of curiousity why not `#include <stdio.h>` always?

Reply:
Um, some desire to avoid namespace pollution? Don't want to force those things into global namespace for apps?

Probably a bit pointless, given that mbed.h has `using namespace mbed` and `using namespace std`.

Reply:
@kjbracey-arm,

> Probably a bit pointless, given that mbed.h has `using namespace mbed` and `using namespace std`.

Hopefully not for long: #5679.